### ISSUE TITLE
Memory leak in .jaccard

### ISSUE THREAD

**Issue by ag14774** [dib-lab/sourmash#685]
```python
def get_average_dist(file_list):
    file_list = [str(f) + '.sig' for f in file_list]
    signatures = [sourmash.load_one_signature(f) for f in file_list]
    print("loaded")
    dist = 0
    counter = 0
    for i in range(len(signatures)):
        for j in range(i + 1, len(signatures)):
            dist += signatures[i].jaccard(signatures[j])
            counter += 1
    return dist / counter
```

With 3375 signatures I try to calculate the distance between all possible combinations of 2 and to calculate the average. As you can see I am not creating a list of all the combinations. Just a list of signatures which takes less than 800 MBs. Then once it enters the loop, memory usage increases very quickly with each iteration. Any ideas what is going wrong here?

**Comment by camillescott** [comment]
There was indeed a memory leak in a function called by `jaccard` -- a heap-allocated object was not being deleted. I put up a PR to fix it.

**Comment by luizirber** [comment]
This is now released in 2.0.1, thanks @camillescott!

**Comment by ctb** [comment]
Fixed in #687.
